doc/ui-guidelines.md

```diff
@@ -0,0 +1,257 @@
+    [rn/view {:style style/container}
+     (case selected-tab
+       :assets       [assets/view]
+       :collectibles [collectibles/view {:collectibles         collectible-list
+                                         :on-collectible-press on-collectible-press}])
+       [activity/view]]))
+```
+
+## Regular atoms
+
+In some cases regular atoms were used in the components
+
+
+BEFORE:
+```clojure
+(defn view
+  []
+  (let [focused? (atom false)]
+    (fn []
+      (let [on-clear               (fn []
+                                     (reset! status (if @focused? :active :default)))
+            on-focus               (fn []
+                                     (reset! focused? true))
+            on-blur                (fn []
+                                     (reset! focused? false))]))))
+```
+
+NOW:
+```clojure
+(defn view
+  []
+  (let [[focused? set-focused] (rn/use-state false)
+        on-clear               (rn/use-callback
+                                (fn []
+                                  (set-status (if focused? :active :default)))
+                                [focused?])
+        on-focus               (rn/use-callback
+                                (fn []
+                                  (set-focused true)))
+        on-blur                (rn/use-callback
+                                (fn []
+                                  (set-focused false)))]))
+```
+
+> [!IMPORTANT]
+> Instead of a regular atom, we also use a state hook. It's important to understand that its value is now encapsulated within the function, and we must pass it as a dependency to `use-callback` so that the function re-executes with the new value.
+
+## Effects
+
+In most cases, the effect should be used for implementing visual reactions (such as animations) to state changes or for lifecycle methods.
+
+ANIMATION:
+```clojure
+(defn f-zoom-button
+  [{:keys [selected? current-zoom]}]
+  (let [size (reanimated/use-shared-value (if selected? 37 25))]
+    (rn/use-effect #(reanimated/animate size (if selected? 37 25)) [current-zoom])
+    [rn/touchable-opacity
+     {:style (style/zoom-button-container size)}]))
+```
+
+This effect will be run only when current-zoom has changed.
+
+LIFECYCLE:
+
+```clojure
+(defn view
+  [{:keys []}]
+  (let [opacity (reanimated/use-shared-value 0)]
+    (rn/use-mount #(reanimated/animate opacity 1))
+    [rn/view
+     {:style (style/opacity opacity)}]))
+```
+
+```clojure
+(defn view
+  [{:keys []}]
+  (let []
+    (rn/use-unmount #(rn/dispatch [:unmounted]))
+    [rn/view]))
+```
+
+> [!IMPORTANT]
+> Effects shouldn't be used as a reaction to state changes for altering logic.
+
+BAD:
+
+```clojure
+(defn view
+  [collectible-list (rf/sub [:wallet/all-collectibles])]
+  (let []
+    (rn/use-effect #(rn/dispatch [:all-collectibles-changed]) [collectible-list])
+    [rn/view]))
+```
+
+Instead `:all-collectibles-changed` shdoul be used in the handler which changes `collectible-list` state
```

Comment:
typo, shdoul

Reply:
I also think it is better to just give the example you are referring to rather than describe it and hope the reader understands it  
i.e we have a BAD and a GOOD example here 👍

Reply:
it's kinda difficult to give good example here